types/wlr_idle.c

```diff
@@ -135,14 +139,29 @@ static void create_idle_timer(struct wl_client *client,
 		wl_resource_post_no_memory(idle_resource);
 		return;
 	}
-	// arm the timer
-	wl_event_source_timer_update(timer->idle_source, timer->timeout);
+	if (timer->enabled) {
+		// arm the timer
+		wl_event_source_timer_update(timer->idle_source, timer->timeout);
+	}
 }
 
 static const struct org_kde_kwin_idle_interface idle_impl = {
 	.get_idle_timeout = create_idle_timer,
 };
 
+void wlr_idle_update_enabled(struct wlr_idle *idle, bool enabled) {
```

Comment:
I'd prefer `set_enabled` (we sort of try to only use `update` for impl → interface calls)

Reply:
All `wlr_output_update_*` are defined in `interfaces/wlr_output.h`, which means they should not be used by compositors, they should only be used by output implementations. `types/wlr_output.h` has some `wlr_output_set_*` functions I believe.